src/data_access/UserRatingAccessObject.java

```diff
@@ -40,4 +32,44 @@ public HashMap<Movie, Integer> getRatings(String user){
         ratings.put(APIcaller.getMovie("tt2935510"), 3); // Ad Astra
         return ratings;
     }
+    @Override
+    public void removeRating(String username, String movie_id){
+        HashMap<Movie, Integer> rating = getRatings(username);
+        rating.remove(movie_id);
```

Comment:
The hashmap you are retrieving is working with Movie, Integer.
You can't pass the movie_id as a key, as the key is the Movie itself.

I would argue that the issue is not here, but it is with the database. The database should not contain a hashmap of Movie to Integer. It should be
imdbID to Integer.

Also while looking at it, it seems that the getRatings call currently 
is returning a List<Movies> not a HashMap.